builder/struct.go

```diff
@@ -39,6 +39,7 @@ func (*Struct) Build(gen Generator, ctx *MethodContext, sourceID *xtype.JenID, s
 	usedSourceID := false
 	for i := 0; i < target.StructType.NumFields(); i++ {
 		targetField := target.StructType.Field(i)
```

Comment:
`matchTag` shouldn't take precedence over name matching. If there is a clash, then goverter should error. E.g. this
```
// goverter:converter
type Converter interface {
	// goverter:matchTag json
	Convertre(source Input) Output
}

type Input struct {
	A int `json:"B"`
	B int `json:"A"`
}

type Output struct {
	A int `json:"B"`
}
```
The error should be similar to this: https://github.com/jmattheis/goverter/blob/main/scenario/match_ignore_case_unresolved_ambiguity.yml

Reply:
This adds complexity, so I hesitated. Again, happy to add it.